### ISSUE TITLE
Crash EXC_BREAKPOINT: BUG IN CLIENT OF LIBDISPATCH: Unbalanced call to dispatch_group_leave()

### ISSUE THREAD

**Issue by nalexn** [GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk#88]
**Describe the bug**

The SDK crashes upon initialization

**Integration Method**

* [x] SPM (iOS)
* [ ] Cocoapods (iOS)
* [ ] Direct Download (iOS)
* [ ] GMaven (Android)

**SDK Version**
18.4.1

**To Reproduce**

Apparently, simply launch the SDK.

**Xcode version for iOS (please complete the following information):**
 - Xcode 15.2

**Device (please complete the following information):**
 - Various devices, mostly on iOS 17

```
EXC_BREAKPOINT: BUG IN CLIENT OF LIBDISPATCH: Unbalanced call to dispatch_group_leave()

0  libdispatch.dylib +0x3a08c      _dispatch_group_leave.cold.1
1  libdispatch.dylib +0x514c       _dispatch_group_leave
2  OPCoreServices +0x8f79c8        specialized rYum3kFX.iSfZOzDK(_:)
3  OPCoreServices +0x8f7a1c        specialized rYum3kFX.iSfZOzDK(_:)
4  OPCoreServices +0x8f37bc        closure #1 in d3q9cvwI.z5Bu9ugi(siteKey:timeout:ci5gxaVB:completionHandler:)
5  OPCoreServices +0x8fee84        partial apply for closure #1 in lzjQEBmQ(_:_:)
6  OPCoreServices +0x3b6b4         thunk for @escaping @callee_guaranteed () -> () (<compiler-generated>)
7  libdispatch.dylib +0x26a4       __dispatch_call_block_and_release
8  libdispatch.dylib +0x42fc       __dispatch_client_callout
9  libdispatch.dylib +0xb890       __dispatch_lane_serial_drain
10 libdispatch.dylib +0xc3c0       __dispatch_lane_invoke
11 libdispatch.dylib +0x17000      __dispatch_root_queue_drain_deferred_wlh
12 libdispatch.dylib +0x16874      __dispatch_workloop_worker_thread
13 libsystem_pthread.dylib +0x1960 __pthread_wqthread
```

**Comment by walterjgsp** [comment]
Hi @nalexn, thanks for the report. What's the version of the SDK that you're using 18.4.1?

**Comment by nalexn** [comment]
Yes, 18.4.1 for iOS

**Comment by walterjgsp** [comment]
Thanks @nalexn, here are some follow up questions that will help us investigate.

Are you seeing this on production or development? Is there any reproduction steps of any sort? We are launching the SDK and even have stress tests but we are not able to reproduce. How often does it happen?

**Comment by nalexn** [comment]
This is production, repro steps are unknown. So far the crash is 100% when the app is in the background. From the logs it looks like the app is launched and minimized in a couple of seconds, then it crashes.

**Comment by walterjgsp** [comment]
@nalexn, do you see this problem only on 18.4.1 or do you have any report of the same happening in 18.4.0?

**Comment by nalexn** [comment]
We did not have these two crashes (#88 and #89) in 18.4.0

**Comment by nalexn** [comment]
This is new in 18.4.1

**Comment by wiyun0518** [comment]
@nalexn Dear nalexn, I am an iOS developer and I took over a project that uses #pod 'RecaptchaEnterprise','17.0.0', :source =&gt; "Https://cpdc-eap.googlesource.com/RecaptchaEnterprise.git", our online environment can be normal login, But I have been unable to download 'RecaptchaEnterprise','17.0.0' for some reason. I can only use 18.4.0, but since February 7 or so, I can not log in the server colleagues to help me check the reason is to prompt my token verification, but the score is very low. Now several of my colleagues can not log in with the package I typed or it is difficult to log in, but they do not log in frequently, I use the OC version, I do not know what the reason is, I would like to consult you, thank you. Looking forward to your reply.

**Comment by wiyun0518** [comment]
@walterjgsp Dear walterjgsp, I am an iOS developer and I took over a project that uses #pod 'RecaptchaEnterprise','17.0.0', :source => "https://cpdc-eap.googlesource.com/RecaptchaEnterprise.git", our online environment can be normal login, But I have been unable to download 'RecaptchaEnterprise','17.0.0' for some reason. I can only use 18.4.0, but since February 7 or so, I can not log in the server colleagues to help me check the reason is to prompt my token verification, but the score is very low. Now several of my colleagues can not log in with the package I typed or it is difficult to log in, but they do not log in frequently, I use the OC version, I do not know what the reason is, I would like to consult you, thank you. Looking forward to your reply.

**Comment by walterjgsp** [comment]
Hi @wiyun0518, could you open a new issue for better tracking?

**Comment by wiyun0518** [comment]
@walterjgsp ok,

**Comment by walterjgsp** [comment]
By https://github.com/GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk/issues/90 this bug should be solved in the version 18.4.2. Please reopen the bug if this issue appears in the new version of the SDK.
